File: backend/entity/Review.go
package entity

import (
	"gorm.io/gorm"
	"time"
)


type Review struct {
	gorm.Model
	Comment    string    `json:"Comment" valid:"required~Comment is required"`
	Rating     float32      `json:"Rating" valid:"required~Rating is required, range(0|5)~Rating must be between 0 and 5"`	
	ReviewDate time.Time `json:"ReviewDate" valid:"required~ReviewDate is required"`

	UserID uint `json:"UserID"`
	User   User `gorm:"foreignKey:UserID"`

	SheetID uint  `json:"SheetID"`
	Sheet   Sheet `gorm:"foreignKey:SheetID"`
}

File: frontend/src/Interface/Review.ts
import { UserInterface } from "./IUser";
import { SheetInterface } from "./Sheet";


export interface ReviewInterface {
  id?: number;            
  comment: string;        
  rating: number;         
  reviewDate?: string;     
  userId: number;         
  user?: UserInterface;  
  sheetId: number;        
  sheet?: SheetInterface; 
}
